ext/liquid_c/block.c

```diff
@@ -56,6 +56,10 @@ static VALUE rb_block_parse(VALUE self, VALUE tokens, VALUE options)
             case TOKEN_RAW:
             {
                 VALUE str = rb_enc_str_new(token.str, token.length, utf8_encoding);
+
+                if(token.trim_whitespace)
+                   rb_funcall(str, rb_intern("lstrip!"), 0);
```

Comment:
Why not find the offset after any whitespace (see [lstrip_offset](https://github.com/ruby/ruby/blob/c9df0f82b7f364e95a9cf8a9492b00759cfe61af/string.c#L7926)) and create the string with that offset?